pageserver/src/pgdatadir_mapping.rs

```diff
@@ -1848,7 +1848,7 @@ impl<'a> DatadirModification<'a> {
                 // We already updated aux files in `self`: emit a delta and update our latest value.
                 dir.upsert(file_path.clone(), content.clone());
                 n_files = dir.files.len();
-                if aux_files.n_deltas == MAX_AUX_FILE_DELTAS {
+                if n_files == 0 || aux_files.n_deltas == MAX_AUX_FILE_DELTAS {
```

Comment:
Why this change?

Reply:
Is this required for this PR, or is this an indepdendent optimization? Maybe open a separate PR for it?